### PR TITLE
Add abilitiy to take screenshots with a key shortcut

### PR DESCRIPTION
**Pull request by dmarcos**
It introduces a scene level component `screenshot` that allows to take a snapshot of the rendered scene with the `alt-S` shortcut. It is configurable to take either equirectangular (panoramas) or screen snapshots. It should also work with the inspector. @chriscar I believe you were asking for this a couple of days ago in Slack. It would be great to get your feedback on it.

I still have to organized the code a little bit better but the functionality is complete. 

- Bumps three.js r81

### CODE REVIEW

**Comment by chriscar** [comment]
Thanks for letting me know, will have a look. But I have to create a screenshot programmatically, not manually.

**Comment by AdaRoseEdwards** [comment]
Something like `document.querySelector('[screenshot]').components.screenshot.capture()` should work @chriscar

File: tests/shaders/standard.test.js [hunk 1]
```diff
@@ -26,7 +26,6 @@ suite('standard material', function () {
     el.setAttribute('material', 'ambientOcclusionMapIntensity: 0.4; ambientOcclusionMap: url(' + imageUrl + ');');
     assert.equal(el.getObject3D('mesh').material.aoMapIntensity, 0.4);
     el.addEventListener('materialtextureloaded', function (evt) {
-      assert.equal(evt.detail.texture.image.getAttribute('src'), imageUrl);
```
Comment (ngokevin) [inline tests/shaders/standard.test.js]: Curious what three.js changed here?

File: tests/components/scene/screenshot.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* global assert, setup, suite, test, sinon */
+suite('screenshot', function () {
+  setup(function (done) {
+    var el = this.sceneEl = document.createElement('a-scene');
+    this.sinon = sinon.sandbox.create();
+    el.addEventListener('loaded', function () { done(); });
+    document.body.appendChild(el);
+  });
+
+  test('capture is called when key shortut is pressed', function () {
```
Comment (ngokevin) [inline tests/components/scene/screenshot.test.js]: typo shortcut

File: docs/components/screenshot.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
+title: screenshot
+type: components
+layout: docs
+parent_section: components
+---
+
+Take screenshots of the scene using a keboard shortcut (alt+s).
+It can be configured to either take equirectangular captures (panoramas) or
```
Comment (ngokevin) [inline docs/components/screenshot.md]:
```
360&deg; captures (`equirectangular`) or regular screenshots (`projection`)
```

File: docs/components/screenshot.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+regular screenshots. The screenshot component only applies to the [`<a-scene>` element][scene].
+
+## Example
```
Comment (ngokevin) [inline docs/components/screenshot.md]: I'd mention that it's a default component

File: docs/components/screenshot.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+<a-entity screenshot="projection: equirectangular"></a-entity>
+```
+
+## Properties
+
+| Property     | Description                                                | Default Value |
+|--------------|------------------------------------------------------------|---------------|
+| width        | The width in pixels of the screnshot taken.                | 4096          |
+| height       | The height in pixels of the screenshot taken.              | 2048          |
+| projection   | The screenshot projection: perspective or equirectangular | perspective   |
```
Comment (ngokevin) [inline docs/components/screenshot.md]: Should `equirectangular` be default? Seems more useful / cooler.

File: src/components/scene/screenshot.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
+/* global ImageData, URL */
+var register = require('../../core/component').registerComponent;
```
Comment (ngokevin) [inline src/components/scene/screenshot.js]: registerComponent

**Comment by ngokevin** [comment]
The perspective screenshot is upside down and reverse

**Comment by dmarcos** [comment]
I made a bunch of changes based on review and fixed the vertically flipped pixels on equirectangular projection.

**Comment by ngokevin** [comment]
There are still a few typo/doc comments to address in the diff.

Also, curious what changed about how three.js stores ``s? Otherwise, r+wc

**Comment by dmarcos** [comment]
I fixed the typos and changed the shortcut to `alt+ctrl+s`. `alt+s` opens the history menu on Firefox. Using `alt+ctrl+s` also makes it consistent with the inspector shortcut

**Pull request merged**